include/swift/Parse/ParseVersion.h

```diff
@@ -0,0 +1,45 @@
+//===--- VersionParser.h - Parser Swift Version Numbers ---------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Basic/Version.h"
+
+namespace swift {
+class DiagnosticEngine;
+
+namespace version {
+/// Returns a version from the currently defined SWIFT_COMPILER_VERSION.
+///
+/// If SWIFT_COMPILER_VERSION is undefined, this will return the empty
+/// compiler version.
+Version getCurrentCompilerVersion();
+} // namespace version
+
+class VersionParser final {
```

Comment:
Is there a reason to make this a class and add `static` methods rather than a namespace and free functions for `parseCompilerVersionString` and `parseVersionString`?

Reply:
I need to friend Version so I can poke at the component vector.